File: src/main/java/io/bindernews/thegrackle/cardmods/DoubleAloftDamageMod.java
package io.bindernews.thegrackle.cardmods;

import basemod.abstracts.AbstractCardModifier;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import io.bindernews.thegrackle.GrackleMod;
import io.bindernews.thegrackle.stance.StanceAloft;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class DoubleAloftDamageMod extends AbstractCardModifier {

    @Override
    public float modifyBaseDamage(float damage, DamageInfo.DamageType type, AbstractCard card, AbstractMonster target) {
        AbstractCreature owner = iop().getCardOwner(card);
        if (StanceAloft.isAloft(owner)) {
            return damage * 2.f;
        } else {
            return damage;
        }
    }

    @Override
    public String modifyDescription(String rawDescription, AbstractCard card) {
        String ln = GrackleMod.miscUI.get().get("double_aloft_damage");
        return String.format("%s NL %s", rawDescription, ln);
    }

    @Override
    public AbstractCardModifier makeCopy() {
        return new DoubleAloftDamageMod();
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cardmods/DoubleAloftDamageMod.kt
package io.bindernews.thegrackle.cardmods

import basemod.abstracts.AbstractCardModifier
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.DamageInfo.DamageType
import com.megacrit.cardcrawl.monsters.AbstractMonster
import io.bindernews.thegrackle.GrackleMod
import io.bindernews.thegrackle.helper.ModInterop
import io.bindernews.thegrackle.stance.StanceAloft

class DoubleAloftDamageMod : AbstractCardModifier() {
    override fun modifyDamage(damage: Float, type: DamageType, card: AbstractCard, target: AbstractMonster): Float {
        val owner = ModInterop.iop().getCardOwner(card)
        return if (StanceAloft.isAloft(owner)) {
            damage * 2f
        } else {
            damage
        }
    }

    override fun modifyDescription(rawDescription: String, card: AbstractCard): String {
        val ln = GrackleMod.miscUI.get()["double_aloft_damage"]
        return String.format("%s NL %s", rawDescription, ln)
    }

    override fun makeCopy(): AbstractCardModifier {
        return DoubleAloftDamageMod()
    }
}